django/db/models/sql/subqueries.py

```diff
@@ -114,6 +114,16 @@ def add_update_values(self, values):
         """
         values_seq = []
         for name, val in values.items():
+            if "__" in name:
+                attributes = name.split('__')
+                name = attributes.pop(0)
+                hierarchy = (',').join(attributes)
+                field = self.get_meta().get_field(name)
+                from django.contrib.postgres.functions import Jsonify
```

Comment:
Importing `contrib.postgres` in `db.models` isn't appropriate. Probably this should be delegated to the database backend. What if the field isn't a JSON field?